### DISCUSSION TITLE
Garbage collection when scope values are Symbols

### DISCUSSION THREAD

**Discussion by iwoplaza** [saasquatch/bunshi#38] [Q&A] [answered]
Hi! I was an example of using scopes for Forms, and the Vue example was providing a symbol as the value of a scope.

```vue
<script setup>
import { provideScope } from 'bunshi/vue';
import { formIdScope } from '../scopes'

const formId = Symbol();
provideScope([formIdScope,formId])

</script>

<template>
  <form>
    Form id: {{formId}}
    <slot></slot>
  <form>
</template>
```

I was wondering if the molecule instance garbage collection is handled by Bunshi in this case, given that the symbol for example is not used by anything else anymore? (the component that provided the scope was unmounted).

**Comment by iwoplaza** [comment]
I personally use React, but I guess this functionality should be framework agnostic, if implemented.

**Comment by loganvolkers** [comment, marked as answer]
Yes, the browser's JavaScript engine should garbage collect the symbols, and the related bunshi molecules, when there are no more references to them.
